torch/nn/init.py

```diff
@@ -365,12 +365,10 @@ def sparse_(tensor, sparsity, std=0.01):
     with torch.no_grad():
         tensor.normal_(0, std)
         for col_idx in range(cols):
-            row_indices = list(range(rows))
-            random.shuffle(row_indices)
+            row_indices = torch.randperm(rows)
             zero_indices = row_indices[:num_zeros]
-            for row_idx in zero_indices:
-                tensor[row_idx, col_idx] = 0
-
+            t = tensor[:, col_idx]
+            t[zero_indices] = 0
```

Comment:
nit: those two lines can be simplified as
```python
tensor[zero_indices, col_idx] = 0
```

Reply:
true, I noticed that as well literally after pushing, going to update now

Reply:
On PyTorch 0.3.1 this is going to raise an indexing error.

Reply:
It might, but pytorch 0.3.1 will have the old `sparse_` implementation

Reply:
Of course I was saying it just in case that the indexing errors were still handled as in 0.3. I didn't know it was "fixed" in 0.4, that's great